src/lib/components/Map/Map.js

```diff
@@ -97,17 +98,29 @@ class Map extends Component {
       this.map.on('zoom', this.handleLabelsOnMapZoom.bind(this))
 
       // Dispach map rendered to indicate map was rendered
-      this.props.dispatch(Actions.mapRendered(mapId));
+      this.props.dispatch(Actions.mapRendered(mapId, true));
     }
   }
 
   addMouseEvents(mapId) {
+    const { handlers } = this.props;
+    if (handlers && Array.isArray(handlers)) {
+      // let handler;
+
+      for (let c = 0; c < handlers.length; c += 1) {
+        this.map.on(handlers[c].type, handlers[c].method);
```

Comment:
@KipSigei - Also it looks like [MapBox has an option to apply events to a single layer](https://docs.mapbox.com/mapbox-gl-js/api/#map#on), maybe we could add an optional `layer` property to the `handler[c]` object, and check for it before subscribing the event.

Reply:
@cKellyDesign sweet! didn't know we could have events subscribed to a specific layer